website/views/shared/tasks/task.jade

```diff
@@ -166,22 +166,56 @@ li(bindonce='list', id='task-{{::task.id}}', ng-repeat='task in obj[list.type+"s
         // if Daily, calendar
         fieldset(ng-if='::task.type=="daily"', class="option-group")
           legend.option-title=env.t('repeat')
-          ul.repeat-days(bindonce)
-            // note, does not use data-toggle="buttons-checkbox" - it would interfere with our own click binding
-            li
-              button(ng-class='{active: task.repeat.su}', type='button', ng-click='task.challenge.id || (task.repeat.su = !task.repeat.su)') {{::moment.weekdaysMin(0)}}
-            li
-              button(ng-class='{active: task.repeat.m}', type='button', ng-click='task.challenge.id || (task.repeat.m = !task.repeat.m)') {{::moment.weekdaysMin(1)}}
-            li
-              button(ng-class='{active: task.repeat.t}', type='button', ng-click='task.challenge.id || (task.repeat.t = !task.repeat.t)') {{::moment.weekdaysMin(2)}}
-            li
-              button(ng-class='{active: task.repeat.w}', type='button', ng-click='task.challenge.id || (task.repeat.w = !task.repeat.w)') {{::moment.weekdaysMin(3)}}
-            li
-              button(ng-class='{active: task.repeat.th}', type='button', ng-click='task.challenge.id || (task.repeat.th = !task.repeat.th)') {{::moment.weekdaysMin(4)}}
+
+          // Choose the frequency
+          ul.priority-multiplier
             li
-              button(ng-class='{active: task.repeat.f}', type='button', ng-click='task.challenge.id || (task.repeat.f= !task.repeat.f)') {{::moment.weekdaysMin(5)}}
+              button(ng-class='{active: task.frequency=="daily"}', type='button', ng-click='task.frequency="daily"')='Daily'
             li
-              button(ng-class='{active: task.repeat.s}', type='button', ng-click='task.challenge.id || (task.repeat.s = !task.repeat.s)') {{::moment.weekdaysMin(6)}}
+              button(ng-class='{active: task.frequency=="weekly"}', type='button', ng-click='task.frequency="weekly"')='Weekly'
+
+          // If frequency is daily
+          fieldset(ng-show='task.frequency=="daily"')
+            | Once every&nbsp;&nbsp;
```

Comment:
Actually mixins aren't really necessary for my idea. Here's a simpler idea to take care of this case: https://github.com/fallenpanda1/habitrpg/commit/cc72c3bccabd0b55f2b4efdf1aef4ed6acc8251f.